### ISSUE TITLE
lint: too many returns

### ISSUE THREAD

**Issue by cristaloleg** [go-critic/go-critic#485]
Name: `tooManyReturns`
When a function has more that 5/7/10 (can be configured by a user) - suggest to simplify a function.
Maybe we should omit switches completely or add flag too.

Suggested by @tenmozes 🙌

**Comment by Quasilyte** [comment]
Can be implemented with `StmtVisitor`.

**Comment by Quasilyte** [comment]
What if only some switch cases do return?
I think switch with return inside every case is somewhat common as a table lookup alternative (LUT), but if only a couple of case bodies do return, they should be counted as ordinary returns since they do make code more hard to follow.

Anyway, we need a better defined set of heuristics.
